docs/migration-guide.asciidoc

```diff
@@ -0,0 +1,311 @@
+`NEST` (v7) which only supported mocking of top-level client methods.
+
+We have also introduced a `TestableResponseFactory` in `Elastic.Transport` to 
+make it easier to create response instances with specific status codes and validity 
+that can be used during unit testing.
+
+These changes are in addition to our existing support for testing with an 
+`InMemoryConnection`, virtualized clusters and with our 
+https://github.com/elastic/elasticsearch-net-abstractions/blob/master/src/Elastic.Elasticsearch.Managed[`Elastic.Elasticsearch.Managed`] library for integration 
+testing against real {es} instances.
+
+[discrete]
+=== Migrating to Elastic.Clients.Elasticsearch
+
+[WARNING]
+--
+The version 8 client does not currently have full-feature parity with `NEST`. The 
+client primary use case is for application developers communitating with {es}.
+-- 
+
+The version 8 client focuses on core endpoints, more specifically for common CRUD 
+scenarios. We intend to reduce the feature gap in subsequent versions. We recommend reviewing this documentation carefully to learn about the missing features and reduced API surface details before migrating from the v7 `NEST` client.
+
+The choice to code-generate a new evolution of the {net-client} introduces some 
+significant breaking changes. We consciously took the opportunity to refactor 
+and reconsider historic design choices as part of this major release, intending 
+to limit future breaking changes.
+
+The v8 client is shipped as a new https://www.nuget.org/packages/Elastic.Clients.Elasticsearch/[NuGet package] 
+which can be installed alongside v7 `NEST`. We  
+anticipate that some consumers may prefer a phased migration with both 
+packages side-by-side for a short period of time to manage complex migrations. In addition, `NEST` 7.17.x can continue to be used in 
+https://www.elastic.co/guide/en/elasticsearch/client/net-api/7.17/connecting-to-elasticsearch-v8.html[compatibility mode] 
+with {es} 8.x servers until the v8 {net-client} features 
+align with application requirements.
+
+[discrete]
+=== Breaking Changes
+
+[WARNING]
+--
+As a result of code-generating a majority of the client types, version 8 of 
+the client includes multiple breaking changes.
+--
+
+We have strived to keep the core foundation reasonably similar, but types emitted 
+through code-generation are subject to change between `NEST` (v7) and the new 
+`Elastic.Clients.Elasticsearch` (v8) package.
+
+[discrete]
+==== Namespaces
+
+We have renamed the package and top-level namespace for the v8 client to 
+`Elastic.Clients.Elasticsearch`. All types belong to this namespace. When 
+necessary, to avoid potential conflicts, types are generated into suitable 
+sub-namespaces based on the https://github.com/elastic/elasticsearch-specification[{es} specification]. Additional `using` directives may be required to access such types 
+when using the {net-client}.
```

Comment:
Usually, we try to avoid using first-person plural pronouns in docs.

```suggestion
The package and top-level namespace for the v8 client have been renamed to 
`Elastic.Clients.Elasticsearch`. All types belong to this namespace. When 
necessary, to avoid potential conflicts, types are generated into suitable 
sub-namespaces based on the https://github.com/elastic/elasticsearch-specification[{es} specification]. Additional `using` directives may be required to access such types 
when using the {net-client}.
```